Yosemite/Yosemite/Tools/POS/PointOfSaleOrderService.swift

```diff
@@ -0,0 +1,219 @@
+
+    init(orderItem: OrderItem) {
+        self.itemID = orderItem.itemID
+        self.productID = orderItem.productID
+        self.quantity = orderItem.quantity
+    }
+}
+
+public protocol PointOfSaleOrderServiceProtocol {
+    /// Syncs order based on the cart.
+    /// - Parameters:
+    ///   - cart: Cart with optional items (product & quantity).
+    ///   - order: Optional latest remotely synced order. Nil when syncing order for the first time.
+    ///   - allProducts: Necessary for removing existing order items with products that have been removed from the cart.
+    /// - Returns: Order from the remote sync.
+    func syncOrder(cart: [PointOfSaleCartItem], order: PointOfSaleOrder?, allProducts: [PointOfSaleCartProduct]) async throws -> PointOfSaleOrder
+
+    /// Updates status of an order and syncs it
+    /// - Parameters:
+    ///   - posOrder: POS order.
+    ///   - status: New order status.
+    /// - Returns: Updated and synced POS order.
+    func updateOrderStatus(posOrder: PointOfSaleOrder, status: OrderStatusEnum) async throws -> PointOfSaleOrder
+
+    /// Creates WOO Order from POS Order.
+    /// - Parameters:
+    ///   - posOrder: Optional POS order.
+    /// - Returns: Order created from posOrder data.
+    func orderFrom(posOrder: PointOfSaleOrder?) -> Order
+}
+
+public final class PointOfSaleOrderService: PointOfSaleOrderServiceProtocol {
+    // MARK: - Properties
+
+    private let siteID: Int64
+    private let ordersRemote: OrdersRemote
+
+    // MARK: - Initialization
+
+    public convenience init(siteID: Int64, credentials: Credentials) {
+        self.init(siteID: siteID, network: AlamofireNetwork(credentials: credentials))
+    }
+
+    public init(siteID: Int64, network: Network) {
+        self.siteID = siteID
+        self.ordersRemote = OrdersRemote(network: network)
+    }
+
+    // MARK: - Protocol conformance
+
+    public func syncOrder(cart: [PointOfSaleCartItem], order posOrder: PointOfSaleOrder?, allProducts: [PointOfSaleCartProduct]) async throws -> PointOfSaleOrder {
+        let initialOrder: Order = orderFrom(posOrder: posOrder)
+
+        let order = updateOrder(initialOrder, cart: cart, allProducts: allProducts)
+        let syncedOrder: Order
+        if posOrder != nil {
+            syncedOrder = try await ordersRemote.updatePointOfSaleOrder(siteID: siteID, order: order, fields: [.items])
+        } else {
+            syncedOrder = try await ordersRemote.createPointOfSaleOrder(siteID: siteID, order: order, fields: [.items, .status])
+        }
+        return PointOfSaleOrder.createFromOrder(syncedOrder)
+    }
+
+    public func updateOrderStatus(posOrder: PointOfSaleOrder, status: OrderStatusEnum) async throws -> PointOfSaleOrder {
+        let order: Order = orderFrom(posOrder: posOrder)
+
+        let syncedOrder: Order = try await ordersRemote.updatePointOfSaleOrder(siteID: siteID, order: order, fields: [.status])
+        print("🟢 [POS] Synced order status: \(order.status)")
```

Comment:
nit: do we need this print? For necessary logging, using `DDLog` would be more consistent with other use cases